wolfcrypt/src/asn.c

```diff
@@ -33071,13 +33100,24 @@ static int ParseCRL_CertList(DecodedCRL* dcrl, const byte* buf,
 
     checkIdx = idx;
     if (GetASNTag(buf, &checkIdx, &tag, sz) == 0 && tag == ASN_INTEGER) {
-        if (GetMyVersion(buf, &idx, &version, sz) < 0)
+        if (GetMyVersion(buf, &idx, &dcrl->version, sz) < 0)
             return ASN_PARSE_E;
+        dcrl->version++;
     }
 
     if (GetAlgoId(buf, &idx, &oid, oidIgnoreType, sz) < 0)
         return ASN_PARSE_E;
 
+    checkIdx = idx;
+    if (GetSequence(buf, &checkIdx, &length, sz) < 0) {
+        return ASN_PARSE_E;
+    }
+#ifdef OPENSSL_EXTRA
+    else {
+        dcrl->issuer = (byte*)GetNameFromDer(buf + idx, WC_ASN_NAME_MAX);
```

Comment:
This is allocating a buffer too large and copying beyond the provided issuers buffer. This needs to use the `length` from above... It should store in `dcrl->issuerSz` and use it. If size doesn't exist in dcrl then it needs added.

Reply:
Fixed with `dcrl->issuerSz`.